packages/lib-utils/src/components/details-page-header/DetailsPageHeader.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  Split,
+  SplitItem,
+  TextContent,
+  Text,
+  TextVariants,
+  Label,
+  DropdownPosition,
+} from '@patternfly/react-core';
+import * as _ from 'lodash';
+import React from 'react';
+import type { K8sResourceCommon } from '../../types/k8s';
+import type { BreadcrumbProp, ActionButtonProp, ActionMenuProps } from './utils';
+import { Breadcrumbs, ActionButtons, ActionMenu } from './utils';
+import '@patternfly/react-styles/css/utilities/Spacing/spacing.css';
+
+export type DetailsPageHeaderProps = {
+  breadcrumbs: BreadcrumbProp[];
+  actionButtons?: ActionButtonProp[];
+  pageHeading?: string;
+  obj?: K8sResourceCommon;
+  pageHeadingLabel?: {
+    name: string;
+    key?: string;
+    icon?: React.ReactNode;
+    href?: string;
+    color?: 'blue' | 'cyan' | 'green' | 'orange' | 'purple' | 'red' | 'grey';
+  };
```

Comment:
@jeff-phillips-18 Thanks for sharing this example! I could add an icon here - this appears to be a fairly common use case. I see icon in Console too. However, over there it is before the heading. The idea behind this component is to provide a level of consistency across apps but I also agree that it should allow for a certain level of customization. 
As for allowing the implementation to pass a ReactNode for the entire heading or for the labels/icons etc, I'd defer to Bryan to comment on whether this would be good from a consistency aspect - @florkbr Could you weigh in please? Thanks!

Reply:
@florkbr @jeff-phillips-18 I've pushed up a commit with an option to include an icon before or after the title in the page heading.